pkg/controller/node/nodecontroller.go

```diff
@@ -368,7 +369,18 @@ func (nc *NodeController) maybeDeleteTerminatingPod(obj interface{}) {
 
 func forcefullyDeletePod(c clientset.Interface, pod *api.Pod) {
 	var zero int64
-	err := c.Core().Pods(pod.Namespace).Delete(pod.Name, &api.DeleteOptions{GracePeriodSeconds: &zero})
+	serverPod, err := c.Core().Pods(pod.Namespace).Get(pod.Name)
+	if kubeerrors.IsNotFound(err) {
+		glog.V(3).Infof("Pod %q (%s) does not exist on the server", serverPod.Name, serverPod.UID)
+		return
+	}
+	// This doesn't entirely avoid the race condition, pod can be deleted and recreated after this check
+	// but before the deletion request sent to the server.
```

Comment:
Hm. Is it worth it to add this? ( @smarterclayton, opinions?)

It'd be better to add an optional UID precondition to delete. Like, add UID to the DeleteOptions type.